### ISSUE TITLE
Min/Max value from the spinner button is not announced when user increment/decrement the value using arrow keys

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4665]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 4.1.2: Name, role, value and 2.1.1 keyboard

**Test Environment:**
- Test URL: https://clarity.design/documentation/datagrid/built-in-filters
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 2020.1

**Steps to reproduce:**
1. Open the test URL.
2. Navigate to any one of the datagrids with filter options having spinner buttons in it.
3. Press the filter button which has spinner button in it.
4. Try to interact with the spinner button by updating the values, using arrow key navigation or tab key navigation.

**Expected result:**
User should be able to interact with the spinner buttons appropriately with proper spoken feedback by screen reader.

**Actual result:**
1. No announcement is made by screen reader when the spinner buttons value is incremented or decremented using arrow keys. **Observation** This issue is not reproducible in Firefox Quantum 68.8.0esr
2. Overall dialog is focused with tab key or arrow key navigation. With arrow key screen reader says "blank" and with tab navigation it says "section".
